apps/src/music/views/Callouts.tsx

```diff
@@ -100,15 +100,30 @@ const Callouts: React.FunctionComponent = () => {
 
   const calloutIds = callout?.id?.split('---');
 
-  const validCallouts: AvailableCallout[] | undefined = calloutIds?.map(
-    calloutId => availableCallouts[calloutId]
-  );
+  const validCallouts: AvailableCallout[] = [];
+
+  calloutIds?.forEach(calloutId => {
+    const splitId = calloutId.split(':');
+    if (splitId.length === 2) {
+      const dataId = splitId[1];
+      validCallouts.push({
+        selector: `.blocklyWorkspace g[data-id="${dataId}"] path`,
+        direction: splitId[0] === 'id-left' ? 'left' : 'up',
+      });
+    } else if (availableCallouts[calloutId]) {
+      validCallouts.push({
+        selector: availableCallouts[calloutId].selector,
+        direction: availableCallouts[calloutId].direction,
+        openToolboxCategory: availableCallouts[calloutId].openToolboxCategory,
+      });
+    }
+  });
```

Comment:
this should be a `map`:

```
const validCallouts: AvailableCallout[] =   calloutIds?.map((bucket, calloutId) => {
    const splitId = calloutId.split(':');
    if (splitId.length === 2) {
      const dataId = splitId[1];
      bucket.push({
        selector: `.blocklyWorkspace g[data-id="${dataId}"] path`,
        direction: splitId[0] === 'id-left' ? 'left' : 'up',
      });
    } else if (availableCallouts[calloutId]) {
      bucket.push({
        selector: availableCallouts[calloutId].selector,
        direction: availableCallouts[calloutId].direction,
        openToolboxCategory: availableCallouts[calloutId].openToolboxCategory,
      });
    }
    return bucket;
  }, []);
```

Reply:
Oh, I made it not a `.map` because when `availableCallouts[calloutId]` is not found, presumably because of a mistake in the authoring, we don't want to add it to the list of `validCallouts`.

Reply:
I typed the wrong function. I meant `reduce` not `map`. 🙄 I even _wrote_ a `reduce` function and just called it a `map`. Whoops. comment is updated

But the `reduce` is still the right thing to do here - it's still only adding it to the list under the same conditions as the `forEach`, so anything invalid is getting filtered out. The `reduce` just lets it happen in a pure function w/o any side effects mutating an external variable.